include/extractor/guidance/toolkit.hpp

```diff
@@ -476,6 +483,77 @@ inline bool hasRoundaboutType(const TurnInstruction instruction)
     return std::find(valid_types, valid_end, instruction.type) != valid_end;
 }
 
+// Public service vehicle lanes and similar can introduce additional lanes into the lane string that
+// are not specifically marked for left/right turns. This function can be used from the profile to
+// trim the lane string appropriately
+//
+// left|throught|
+// in combination with lanes:psv:forward=1
+// will be corrected to left|throught, since the final lane is not drivable.
+// This is in contrast to a situation with lanes:psv:forward=0 (or not set) where left|through|
+// represents left|through|through
+inline std::string
+trimLaneString(std::string lane_string, std::int32_t count_left, std::int32_t count_right)
```

Comment:
since recently (@danpat and my late night debugging session) it has a dedicated cucumber test for its edge-cases